pulpcore/pulpcore/app/serializers/base.py

```diff
@@ -189,6 +189,40 @@ def _validate_relative_path(self, path):
 
         return path
 
+    def to_representation(self, instance):
+        """
+        Represent a cast Detail instance as a dict of primitive datatypes
+        """
+
+        # This is very similar to DRF's default to_representation implementation in
+        # ModelSerializer, but checks whether a 'minimal_fields' attribute is defined,
+        # and uses it when many objects are being serialized at once (i.e. many=True,
+        # when the viewset is performing a 'list' action).
+        ret = OrderedDict()
+        fields = self._readable_fields
+        if self.parent and self.parent.many and hasattr(self.Meta, 'minimal_fields'):
+            fields = [field for field in fields if field.field_name in self.Meta.minimal_fields]
+
+        for field in fields:
+            try:
```

Comment:
This looks like a good job for [suppress](https://docs.python.org/3.5/library/contextlib.html#contextlib.suppress)